server/src/test/java/org/elasticsearch/cluster/coordination/JoinTaskExecutorTests.java

```diff
@@ -87,9 +87,10 @@ public void testPreventJoinClusterWithUnsupportedNodeVersions() {
             });
         }
 
-        Version oldMajor = Version.V_6_4_0.minimumCompatibilityVersion();
-        expectThrows(IllegalStateException.class, () -> JoinTaskExecutor.ensureMajorVersionBarrier(oldMajor, minNodeVersion));
-
+        if (minNodeVersion.onOrAfter(Version.V_8_0_0)) {
+            Version oldMajor = Version.V_7_2_0.minimumCompatibilityVersion();
```

Comment:
Why is this changed to a hardcoded 7.2 compat version?

Reply:
this was a mis-understanding of the code. Since it stopped things from failing, I thought it was right.

the reason it is confusing is because of how `minimumCompatibilityVersion` is calculated